### PR TITLE
CFY-2209 Host pool service blueprints

### PR DESCRIPTION
**Pull request by msoczewka**

### CODE REVIEW

File: host-pool-service-blueprints/scripts/tasks.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    if not _is_installed('python-dev', _DPKG):
+        _install_with_apt_get('python-dev')
+    if not _is_installed('pip', _WHICH):
+        _install_with_apt_get('python-pip')
+        fabric.api.sudo('pip install --upgrade pip')
+    if not _is_installed('virtualenv', _WHICH):
+        fabric.api.sudo('pip install virtualenv')
+
+
+_DPKG = 1
```
Comment (idanmo) [inline host-pool-service-blueprints/scripts/tasks.py]: Please move these constants to the top of the file (a long with _HOST_POOL_REPOSITORY_LINK)

File: host-pool-service-blueprints/simple-manager-with-host-pool.yaml [hunk 2]
```diff
@@ -0,0 +1,233 @@
+tosca_definitions_version: cloudify_dsl_1_0
```
Comment (idanmo) [inline host-pool-service-blueprints/simple-manager-with-host-pool.yaml]: Latest DSL version is cloudify_dsl_1_1

File: host-pool-service-blueprints/scripts/tasks.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    elif check_method == _WHICH:
+        result = fabric.api.run(
+            'which {0} 1>/dev/null 2>&1'.format(what),
+            quiet=True)
+    else:
+        raise Exception('bad parameter')
+    return result.succeeded
+
+
+_updated = False
```
Comment (idanmo) [inline host-pool-service-blueprints/scripts/tasks.py]: Please move to the top of the file...

**Comment by msoczewka** [comment]
Since this is still open, key file support has been added here and the Host pool service type has been cut out from blueprints.

File: host-pool-service-blueprints/host-pool-service.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+node_types:
+  cloudify.nodes.HostPoolService:
+    derived_from: cloudify.nodes.SoftwareComponent
+    properties:
+      directory:
+        type: string
+        default: /home/ubuntu/host_pool
+      environment:
+        # Accepted environment variables:
```
Comment (iliapolo) [inline host-pool-service-blueprints/host-pool-service.yaml]: this should be in the 'description' of the property, and not commented.

File: host-pool-service-blueprints/host-pool-service.yaml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      ssh_user:
+        type: string
+        default: ubuntu
+      ssh_key_path:
+        type: string
+        default: /home/ubuntu/.ssh/id_rsa
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create:
+          implementation: fabric.fabric_plugin.tasks.run_task
```
Comment (iliapolo) [inline host-pool-service-blueprints/host-pool-service.yaml]: the mapping should be to the script plugin, not the fabric plugin.
that is
create: scripts/tasks.py

**Pull request merged**